cmd/lnd/main.go

```diff
@@ -14,7 +14,16 @@ func main() {
 	// function will also set up logging properly.
 	loadedConfig, err := lnd.LoadConfig()
 	if err != nil {
-		_, _ = fmt.Fprintln(os.Stderr, err)
+		switch flagsErr := err.(type) {
```

Comment:
We should only exit with a non-`0` code if there really was an error (as the `--help` flag isn't really an unsuccessful termination).
In other projects we use the following block:

```
	if e, ok := err.(*flags.Error); err != nil &&
		(!ok || e.Type != flags.ErrHelp) {

		_, _ = fmt.Fprintln(os.Stderr, err)
		os.Exit(1)
	}
```

But I'm fine with the more verbose version as long as the `os.Exit(1)` is moved along with the `Fprintln`.

Reply:
In the end I settled with the shorter version (although imho it's a bit more difficult to read).
PTAL